lib/memoizable/memory.rb

```diff
@@ -47,12 +49,13 @@ def [](name)
     #
     # @api public
     def []=(name, value)
```

Comment:
@dkubb Do we even have a use case for this method?

Reply:
@mbj lemme remove it and see what breaks :) If it's only it's tests then I will remove.

Reply:
@mbj we use this method in quite a few places both within the code and tests and in dependent gems.

I did find one unnecessary use of Memory in the ModuleMethod module, which I removed.